modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/commerce_marketplace_payment_method/MarketplaceViews.tsx

```diff
@@ -37,6 +38,18 @@ export default function MarketplaceViews() {
 		[product]
 	);
 
+	const hasPermissionToInstallApp = (product: MarketplaceProduct) => {
+		if (permissions.canPurchaseAndInstallPaidApps) {
+			return true;
+		}
+
+		if (product.getPriceModel() === 'free') {
+			return permissions.canInstallFreeApps;
+		}
+
+		return false;
+	};
+
```

Comment:
Move this logic to MarketplaceProduct

```js
if (permissions.canPurchaseAndInstallPaidApps) {
    return true;
}

if (marketplaceProduct.getPriceModel() === 'free') {
	return permissions.canInstallFreeApps;
}

return false;
```



```suggestion
	const getButtonAdditionalProps = (marketplaceProduct: MarketplaceProduct) => {
	    
	      if (isProductInstalled()) {
	         return {
	         disabled: true, title: "installed..."}
	        }
	    
	       if (!marketplaceProduct.canInstallApp(permissions)) {
	              return {
	         disabled: true, title: "installed..."}
	        } 
	       }
	};

```